tests/unit/packet/PublishTest.php

```diff
@@ -155,106 +157,45 @@ public function testSetQos()
         );
     }
 
-    public function testSetDup()
```

Comment:
I removed the tests where private properties were being tested. `testPublishStandardWithDup()` and `testParseWithDup()` both test that feature without having to know the specific implementation.